services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/Domain.kt

```diff
@@ -87,6 +87,18 @@ data class TaydennysAttachmentMetadata(
             taydennysId = taydennysId,
         )
     }
+
+    fun toAlluAttachment(content: ByteArray): AlluAttachment =
```

Comment:
To avoid copy-pasting, you could define an interface for AttachmentMetadata with an attachment type, like:
```kotlin
sealed interface AttachmentMetadataWithType : AttachmentMetadata {
    val attachmentType: ApplicationAttachmentType
```
Then if both `ApplicationAttachmentMetadata` and `TaydennysAttachmentMetadata` implement the new interface, `toAlluAttachment()` can be moved to the interface.

Reply:
Good suggestion, thank you! I did just that.